### DISCUSSION TITLE
Call REST APi, PnPJs or Add  PnP Controls in React Component from Custom Web Component

### DISCUSSION THREAD

**Discussion by soumyadeep2207** [microsoft-search/pnp-modern-search#1742] [General]
Hi @FranckyC @wobba ,

AI would like to have your view on below scenarios

How to consume SharePoint REST API/pnp Js from custom react component which is being rendered from  BaseWebComponent
public async connectedCallback() {

 let props = this.resolveAttributes();
 **const debugView = <CustomComponent {...props}/>;**
 ReactDOM.render(debugView, this);
}

If yes then how to pass the context in order to use any PnP Reusable controls like PeoplePicker. which reside in custom react component. 
Reason I am asking this, it may happen that we may call one react components which is rendering a PnP People Picker and this control expects context object coming from props.

Looking forward to hear from you. 
Thanks in advance

**Comment by FranckyC** [comment]
Hi @soumyadeep2207 , Web Part context is not available in the web components for now. You can only use common shared services using the default service scope.
